c++/CMakeLists.txt

```diff
@@ -79,39 +77,70 @@ else()
   #
   # We have to use -std=gnu++0x isntead of -std=c++11 because otherwise we lose
   # GNU extensions that we need.
-  set(CMAKE_CXX_FLAGS "-std=gnu++0x -Wall -Wextra -Wno-strict-aliasing -Wno-sign-compare -Wno-unused-parameter -pthread ${CAPNP_LITE_FLAG} ${CMAKE_CXX_FLAGS}")
+  add_compile_options(-std=gnu++0x -Wall -Wextra -Wno-strict-aliasing -Wno-sign-compare -Wno-unused-parameter -pthread)
```

Comment:
It needs to be passed to `kj` as well because of [this check in `kj/common.h`](https://github.com/sandstorm-io/capnproto/blob/f6a31613efce413795a200ccc194c2c05d463da2/c%2B%2B/src/kj/common.h#L67).